### PR TITLE
Refactor Api Structure

### PR DESCRIPTION
**Pull request by ps3611**
Created an API service and added Nestoria API client

### CODE REVIEW

File: api/class/nestoria.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    const response = await fetch(this.endpoint, {
+      method: 'GET',
+    });
+    const responseJson = await response.json();
+    return responseJson.response.listings;
+  }
+
+  processData(rawHomesArray) {
+    const processedHomesArray = rawHomesArray.map((obj) => {
+      const processedObj = {};
```
Comment (stevenvana) [inline api/class/nestoria.js:21]: can you do this with destructuring syntax?

**Review by stevenvana — APPROVED** [review]

File: api/class/nestoria.js (hunk 1 above)
Comment (ps3611) [reply api/class/nestoria.js:21]: there is definitely a better way of doing this! im not sure destructuring is the way though..
ill improve in the next version

Comment (arol) [reply api/class/nestoria.js:21]: Well, yes… and no. It's not easy to refactor this. I think is your time to step into serializers. Serializers are classes/methods that format objects, usually between input and database. In this case you're receiving everything from an input (api) and storing it into the database.

In there you can destructure and structuring again but more lines, maybe more confusion… you can try it.

Comment (ps3611) [reply api/class/nestoria.js:21]: nice nice! ive heard so much about serializers, so now ill look into it!

**Pull request merged**